src/steps/MatchColumnsStep/utils/setColumn.ts

```diff
@@ -9,11 +9,17 @@ export const setColumn = <T extends string>(
 ): Column<T> => {
   switch (field?.fieldType.type) {
     case "select":
+      const matchedOptions = uniqueEntries(data || [], oldColumn.index)?.map(option => {
+        const value = options.find(o => o.value == option.value || o.label == option.entry)?.value
+        return value ? {...option, value} as MatchedOptions<T>  : option as MatchedOptions<T>
+      })
+      const allMatched = matchedOptions.filter(o => o.value).length == options.length
```

Comment:
Some tweaks: 
- Fix: `options` I assume is `field.fieldType.options`
- Changes comparing the length against the full set of unique entries

```suggestion
      const uniqueData = uniqueEntries(data || [], oldColumn.index)
      const matchedOptions = uniqueData?.map(option => {
        const value = field.fieldType.options.find(o => o.value == option.value || o.label == option.entry)?.value
        return value ? {...option, value} as MatchedOptions<T>  : option as MatchedOptions<T>
      })
      const allMatched = matchedOptions.filter(o => o.value).length == uniqueData?.length
```

Reply:
@Yo1L any chance you could plop these tweaks in? Our release is blocked on this getting fixed. Would do it myself but don't want to take the credit

Reply:
Hello everybody, @Yo1L any chance to update the component with this fix? It will be awesome and very welcome! Today the match neither works with value or label, it's a problem...
Thank you very much!